fpga/Makefile

```diff
@@ -0,0 +1,124 @@
+#########################################################################################
+# fpga prototype makefile
+#########################################################################################
+
+#########################################################################################
+# general path variables
+#########################################################################################
+base_dir=$(abspath ..)
+sim_dir=$(abspath .)
+
+# do not generate simulation files
+sim_name := none
+
+#########################################################################################
+# include shared variables
+#########################################################################################
+include $(base_dir)/variables.mk
+
+# default variables to build the arty example
+SUB_PROJECT       := fpga
+SBT_PROJECT       := freedomPlatforms
+MODEL             := E300ArtyDevKitFPGAChip
+VLOG_MODEL        := E300ArtyDevKitFPGAChip
+MODEL_PACKAGE     := sifive.freedom.everywhere.e300artydevkit
+CONFIG            := E300ArtyDevKitConfig
+CONFIG_PACKAGE    := sifive.freedom.everywhere.e300artydevkit
+GENERATOR_PACKAGE := chipyard
+TB                := none # unused
+TOP               := E300ArtyDevKitPlatform
+
+# setup the board to use
+BOARD ?= arty
+
+#########################################################################################
+# misc. directories
+#########################################################################################
+bootrom_dir := $(base_dir)/fpga/bootrom/xip
+fpga_common_script_dir := $(FPGA_DIR)/common/tcl
+fpga_dir := $(base_dir)/fpga/fpga-shells/xilinx
+
+#########################################################################################
+# import other necessary rules and variables
+#########################################################################################
+include $(base_dir)/common.mk
+
+#########################################################################################
+# copy from other directory
+#########################################################################################
+all_vsrcs := \
+	$(sim_vsrcs) \
+	$(base_dir)/generators/sifive-blocks/vsrc/SRLatch.v \
```

Comment:
Why these included via the standard firrtl method?

Reply:
These are just black boxes and do not have a resource associated with them. So we would have to make a PR for these to be included through the "normal automatic" way.

Reply:
For now, I just blindly included them, later I think a PR should be opened in both places to make this automatic.

Reply:
Issue to track?